### PR TITLE
docs: add differentiability warning for Warp with mode='nearest'

### LINKED ISSUE #8540: Add warning to documentation that warp in nearest neighbor mode is non differentiable
**Is your feature request related to a problem? Please describe.**
I spend several days figuring out why my spatial transformer network using a warp layer did not properly train. In the end, the reason is that when using the nearest neighbor mode for the warp layer, the result is non-differentiable.

Pytorch does not give any warnings of this and it is also not mentioned in the documentation of grid_sample.

**Describe the solution you'd like**
An addition to the documentation of the warp layer that warns users of the non-differentiability of the nearest neighbor mode. 

**Describe alternatives you've considered**
I do not really see any other alternatives, as this modification is so small.

### PR DESCRIPTION
**Pull request by Kheil-Z**
Fixes #8540 .

### Description

This PR updates the docstring of monai.networks.blocks.Warp to clarify that mode='nearest' results in zero gradients almost everywhere, effectively blocking gradient flow in registration training.

### Changes:

Added a note to the Warp docstring about the gradient behavior of mode='nearest'.

Added a reference to the semi-supervised registration tutorial:
https://github.com/Project-MONAI/tutorials/blob/main/3d_registration/learn2reg_oasis_unpaired_brain_mr.ipynb

### Next Steps (future PR):

Extend VoxelMorph.forward to handle optional label and keypoint arguments for consistent warping behavior.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Review by ericspod — APPROVED** [review]
Hi @Kheil-Z thanks for the contribution! I think we're good to merge if @KumoLiu could trigger blossom later.
